dev/planning-example.py

```diff
@@ -0,0 +1,292 @@
+
+  This means methods that can take dates can either take astropy
+  dates (as used internally) or python datetime objects (not naive, but
+  tagged with timezone).  So dates can be passed in as timezone-aware
+  datetime objects and results can be easily converted between such.
+
+"""
+
+# ================
+# Observer objects
+# ================
+"""
+An observer defines a single terrestrial location and applicable
+environmental attributes thereof that affect observation calculations
+at that location.
+"""
+# Define an observer by full specification.
+# Latitude/Longitude take input as strings or quantities,
+# elevation, pressure, temperature must be quantities.
+# Accept `timezone` from pytz
+from astroplan import Observer
+import astropy.units as u
+import pytz
+
+obs = Observer(name='Subaru Telescope',
+               longitude='-155:28:48.900',
+               latitude='+19:49:42.600',
+               elevation=4163 * u.meter,
+               pressure=0.615 * u.bar,
+               relative_humidity=0.11,
+               temperature=0 * u.deg_C,
+               timezone=pytz.timezone('US/Hawaii'),
+               description="Subaru Telescope on Mauna Kea, Hawaii")
+
+# Define the same observer with an instance of astropy.coordinates.EarthLocation
+from astropy.coordinates import EarthLocation
+obs = Observer(name='Subaru',
+               location=EarthLocation(lon='-155:28:48.900',
+                                      lat='+19:49:42.600'),
+               elevation=4163 * u.meter,
+               pressure=0.615 * u.bar,
+               relative_humidity=0.11,
+               temperature=0 * u.deg_C,
+               timezone=pytz.timezone('US/Hawaii'),
+               description="Subaru Telescope on Mauna Kea, Hawaii")
+
+# longitude and latitude would be represented internally by astropy
+# angles.Longitude/Latitude.  So any value that can naturally initialize
+# one of those objects could be passed as the parameter to Observer
+
+# It's also possible to initialize using astropy objects and other
+# direct entities
+from astropy.coordinates.angles import Longitude, Latitude
+
+obs = Observer(name='Keck 1',
+               longitude=Longitude(155.4750 * u.degree),
+               latitude=Latitude(19.8264 * u.degree),
+               elevation=4160 * u.m,
+               pressure=0.615 * u.bar,
+               relative_humidity=0.11,
+               temperature=0.0 * u.deg_C,
+               timezone=pytz.timezone('US/Hawaii'),
+               description="W.M. Keck 1 Telescope on Mauna Kea, Hawaii")
+
+# It would also be desirable to be able to have a small database of
+# common telescopes.  Maybe this can at first simply take the form of
+# a python module:
+from astroplan import sites
+
+obs = sites.Keck1
+
+# Environmental conditions should be updatable.  Suggest using keyword
+# parameters (this would make it easy to update an internal AltAz coordinate):
+
+obs.set_environment(pressure=0.600 * u.bar, relative_humidity=0.2,
```

Comment:
Sidestepping the conversation about how it will actually be implemented, it's worth noting that the astropy [Coding Guidelines recommend avoiding the get/set methods](https://astropy.readthedocs.org/en/v1.0.2/development/codeguide.html#coding-style-conventions)

> Classes should either use direct variable access, or python’s property mechanism for setting object instance variables. get_value/set_value style methods should be used only when getting and setting the values requires a computationally-expensive operation.